eng/native/init-compiler-and-cmake.cmd

```diff
@@ -2,6 +2,9 @@
 rem
 rem This file locates VS C++ compilers and cmake for windows.
 
+rem Make sure the current directory stays intact
+set VSCMD_START_DIR="%CD%"
```

Comment:
If this change is applicable repo-wide, it would require additional update of `src\coreclr\build-runtime.cmd`, `src\libraries\Native\build-native.cmd` and `src\installer\corehost\build.cmd` files.

In case it is unclear, this script is only used by mono and IMHO, should be moved next to mono.proj until someone work on https://github.com/dotnet/runtime/pull/44976#discussion_r529633344 (i.e. script has more than one consumer). As normally, only those scripts go in this directory which are shared across the subsets.

Reply:
I think this script is a good start of the shared copy that the rest should be switched to. I agree that there is more work to do.